### ISSUE TITLE
500 error on auth when using gitlab third-party login

### ISSUE THREAD

**Issue by jcruzdesignpdx** [zammad/zammad#4609]
### Used Zammad Version

5.4.1-1683779951.9554bea4.focal

### Environment

- Installation method: deb-package
- Operating system Ubuntu 20.04
- Database + version:  PostgreSQL 12.14
- Elasticsearch version: 7.17.10
- Browser + version: Chrome Version 112.0.5615.137

[production.log](https://github.com/zammad/zammad/files/11466286/production.log)

### Actual behaviour

After a package upgrade, when attempting to login with the gitlab integration a 500 error is thrown. Looking at the production logs I see this Fatal error:

```
I, [2023-05-12T16:27:48.596090#519-492960]  INFO -- : Started POST "/auth/gitlab" for XX.XX.XXX.XX at 2023-05-12 16:27:48 +0000
I, [2023-05-12T16:27:48.599801#519-492960]  INFO -- : (gitlab) Request phase initiated.
I, [2023-05-12T16:27:48.812887#519-440620]  INFO -- : Started GET "/auth/gitlab/callback?code=[FILTERED]&state=e8cc96c4dac02b12d127c86cae02005003d634ddd65c23e3" for XX.XX.XXX.XX at 2023-05-12 16:27:48 +0000
I, [2023-05-12T16:27:48.816586#519-440620]  INFO -- : (gitlab) Callback phase initiated.
F, [2023-05-12T16:27:49.072333#519-440620] FATAL -- :
[c62c12fc-2009-4e12-b9bd-3b1b6b4ad7e8] NoMethodError (undefined method `[]' for nil:NilClass

      uid { raw_info['id'].to_s }
                    ^^^^^^):
[c62c12fc-2009-4e12-b9bd-3b1b6b4ad7e8]
```

It seem like this issue happened a while back too per this issue https://github.com/zammad/zammad/issues/2595

### Expected behaviour

Ability to Login in with Gitlab to zammad

### Steps to reproduce the behaviour

Unknown.

### I'm sure this is a bug and no feature request or a general question.

yes

**Comment by dominikklein** [comment]
Hey @jcruzdesignpdx 

thanks for the report, we have some questions to identify the possible problem.

- From which source version was the upgrade?
- Can you show the used URL for the GitLab third-party login (feel free to remove possible sensitive data)?

**Comment by jcruzdesignpdx** [comment]
hi @dominikklein , the upgrade was from 5.4.1-1682496131.d4dddcd0.focal to 5.4.1-1683779951.9554bea4.focal. The gitlab installation is self-hosted and for the endpoint we use: `https://gitlab.XXXXXXXX.XXX/api/graphql`

**Comment by dominikklein** [comment]
The used URL looks strange, this is normally used in the GitLab Integration for Tickets and not for authentication.

Can you maybe check the content from the following command?

`zammad run rails r "pp Setting.get('auth_gitlab_credentials')"`

**Comment by jcruzdesignpdx** [comment]
@dominikklein this is what I get:
```
{"app_id"=>"1dec3b1577f81cd5096777138c707da0fa012eb19159af9282c0ec506fcd5f7a",
 "app_secret"=>"[redacted]",
 "site"=>"[redacted - this is the domain name where we have our gitlab installation]"}
```

**Comment by dominikklein** [comment]
Looks good on the first view. In the end, there was not a "real" change in the GitLab authentication integration. Only the default placeholder was changed and instances that are using the old "site" style (`api/v4/user` needs to be removed). But when it was working for you before the last update, it should already be in the correct shape, otherwise, it would fail.
I can also not reproduce it with our GitLab instance.
Is there any additional information in the log? Is it happening for every user? Was there some change on your GitLab instance?
With the current information, it seems to be more like a technical question.

**Comment by jcruzdesignpdx** [comment]
Hi @dominikklein , I was not the person who originally set up zammad, however, upon further inspection I found the culprit, I don't know if this was a hiccup in the server upgrade but I realized that the gitlab authentication url had been misconfigured. it was not using https://{git_host}/api/v4/ but rather https://{git_host}// (unsure how that happened after a pkg upgrade). In any case, all seems to be working fine now. Thank you for your patience.

**Comment by dominikklein** [comment]
@jcruzdesignpdx Nice that it's working again. But it's very strange that you need to use `https://{git_host}/api/v4/`, because this suffix `api/v4/` is no longer supported by the used Ruby-Gem (`omniauth-gitlab`) in the code. So because of that during the package upgrade, this suffix was removed from the configuration.

**Comment by GeoffreyFrogeye** [comment]
It seems that the Debian package still ships with an old version of `omniauth-gitlab` (2.0.0 according to `dpkg -S zammad | grep omniauth-gitlab`) while Zammad itself ships with [the migration](https://github.com/zammad/zammad/commit/31bb4528f862ce3502736b858676fba3f0ab1484) for [newer versions](https://github.com/linchus/omniauth-gitlab/commit/2a2549ff2ab314262e932c3ef09c807bc6002c4e) of `omniauth-gitlab`. Tested with version `5.4.1-1684779241.4e46055a.bullseye`.

We got the same issue (same logs too) on May 10th and fixed it the same way (re-adding `/api/v4` that the migration removed), for both our testing and production instance. I don't recall if the code was a 500, the error page was a raw template like on https://github.com/zammad/zammad/issues/3912, but it's probably something else, as I don't recall Zammad ever templating an error page correctly 😅.
